### PR TITLE
Tenant controller logic - only call create network when necessary

### PR DESCRIPTION
**Pull request by Sindica**
Previous tenant controller resync every 5 min. Each time it will try to create arktos network and silently fail if object already exists.

The correct logic is check existence from informer lister before create action.

Before change, API server get POST arktos.futurewei.com request from each tenant every 5 min:
```
ubuntu@ip-172-30-0-14:~/go/src/k8s.io/arktos$ cat /tmp/kube-apiserver0.log | grep POST | grep arktos.futurewei.com | grep aaa
I0121 02:53:02.732248   10439 wrap.go:47] POST /apis/arktos.futurewei.com/v1/tenants/aaa/networks: (4.197716ms) 201 [hyperkube/v0.9.0 (linux/amd64) kubernetes/$Format/system:serviceaccount:kube-system:tenant-controller 172.30.0.14:60436]
I0121 02:55:47.761874   10439 wrap.go:47] POST /apis/arktos.futurewei.com/v1/tenants/aaa/networks: (2.726653ms) 409 [hyperkube/v0.9.0 (linux/amd64) kubernetes/$Format/system:serviceaccount:kube-system:tenant-controller 172.30.0.14:60436]
I0121 03:00:47.763127   10439 wrap.go:47] POST /apis/arktos.futurewei.com/v1/tenants/aaa/networks: (3.632425ms) 409 [hyperkube/v0.9.0 (linux/amd64) kubernetes/$Format/system:serviceaccount:kube-system:tenant-controller 172.30.0.14:60436]
I0121 03:05:47.764969   10439 wrap.go:47] POST /apis/arktos.futurewei.com/v1/tenants/aaa/networks: (5.367067ms) 409 [hyperkube/v0.9.0 (linux/amd64) kubernetes/$Format/system:serviceaccount:kube-system:tenant-controller 172.30.0.14:60436]
I0121 03:10:47.763120   10439 wrap.go:47] POST /apis/arktos.futurewei.com/v1/tenants/aaa/networks: (3.225313ms) 409 [hyperkube/v0.9.0 (linux/amd64) kubernetes/$Format/system:serviceaccount:kube-system:tenant-controller 172.30.0.14:60436]
I0121 03:15:47.763202   10439 wrap.go:47] POST /apis/arktos.futurewei.com/v1/tenants/aaa/networks: (2.916273ms) 409 [hyperkube/v0.9.0 (linux/amd64) kubernetes/$Format/system:serviceaccount:kube-system:tenant-controller 172.30.0.14:60436]
I0121 03:20:47.763869   10439 wrap.go:47] POST /apis/arktos.futurewei.com/v1/tenants/aaa/networks: (3.472108ms) 409 [hyperkube/v0.9.0 (linux/amd64) kubernetes/$Format/system:serviceaccount:kube-system:tenant-controller 172.30.0.14:60436]
I0121 03:25:47.764305   10439 wrap.go:47] POST /apis/arktos.futurewei.com/v1/tenants/aaa/networks: (3.627087ms) 409 [hyperkube/v0.9.0 (linux/amd64) kubernetes/$Format/system:serviceaccount:kube-system:tenant-controller 172.30.0.14:60436]
I0121 03:30:47.765303   10439 wrap.go:47] POST /apis/arktos.futurewei.com/v1/tenants/aaa/networks: (4.444155ms) 409 [hyperkube/v0.9.0 (linux/amd64) kubernetes/$Format/system:serviceaccount:kube-system:tenant-controller 172.30.0.14:60436]

```

After change, there is only one POST arktos.futurewei.com request for each tenant

This PR fix issue https://github.com/CentaurusInfra/arktos/issues/1256

### CODE REVIEW

File: pkg/controller/tenant/tenant_controller.go [hunk 1]
```diff
@@ -239,22 +248,22 @@ func (tc *TenantController) syncTenant(tenantName string) (err error) {
 	}
 
 	// handling the addition of a tenant
-	if err, done := tc.createNamespaces(tenantName); !done {
+	if err, done := tc.syncNamespaces(tenantName); !done {
```
Comment (yb01) [inline pkg/controller/tenant/tenant_controller.go:251]: i guess this is originally based on the namespace controller code, hence the naming of the function as createNamespaces(). they should be changed to syncTenants() in the tenant controller.

File: pkg/controller/tenant/tenant_controller.go [hunk 2]
```diff
@@ -266,23 +275,30 @@ func (tc *TenantController) createDefaultNetworkObject(tenantName string) (error
 	} else {
 		klog.Infof("creating the default network in tenant %q", tenantName)
 		defaultNetwork := arktosv1.Network{}
-		if err := tc.getDefaultNetwork(tenantName, &defaultNetwork); err != nil {
+		if err := tc.constructDefaultNetwork(tenantName, &defaultNetwork); err != nil {
 			failures = append(failures, err)
 		} else {
-			if _, err = tc.networkClient.ArktosV1().NetworksWithMultiTenancy(tenantName).Create(&defaultNetwork); err != nil && !errors.IsAlreadyExists(err) {
+			switch _, err = tc.networkLister.NetworksWithMultiTenancy(tenantName).Get(defaultNetwork.Name); {
```
Comment (yb01) [inline pkg/controller/tenant/tenant_controller.go:281]: is the default network for a tenant immutable? i.e. if a tenant want to change their default network settings, will this be done by the the TC or other ways ?

**Review by yb01 — APPROVED** [review]
/lgtm

File: pkg/controller/tenant/tenant_controller.go (hunk 2 above)
Comment (Sindica) [reply pkg/controller/tenant/tenant_controller.go:281]: I think this was a preliminary design. It was never used before until mizar was introduced. This is creating default network solution for tenant. For now, kubectl can be used to update the default network solution for selected tenant(s). As long as the name is not changes, the same network object will be used for tenant default network solution.

**Pull request merged**